apps/store/themes/store/js/categorization.js

```diff
@@ -0,0 +1,261 @@
+                if(param.indexOf(",") > -1){
+                    innerParams = param.split(",");
+                } else {
+                    innerParams = param.split("%2C");
+                }
+                for(var j= innerParams.length -1; j >= 0; j -= 1){
+                    if(innerParams[j].indexOf(key) > -1){
+                        return decodeURIComponent(innerParams[j]).split(":")[1];
+                    }
+                }
+            }
+        }
+    };
+
+    var formatSearchQuery = function(query){
+        var searchQuery = "";
+        var qjson = JSON.parse('{' + query + '}');
+        var searchKeys = Object.keys(qjson);
+        if ((searchKeys.length === 1) && (searchKeys.indexOf("name") >= 0)) {
+            searchQuery += qjson[searchKeys.pop()];
+        }
+        else {
+            for (var keyIndex in searchKeys) {
+                var key = searchKeys[keyIndex];
+                var value = qjson[key];
+                searchQuery += key + ":" + value + " ";
+            }
+        }
+        searchQuery = searchQuery.trim();
+        return searchQuery;
+    };
+
+    var loadAssets = function(url){
+        $('.assets-container section .ctrl-wr-asset').remove();
+        history.pushState("", "", url);
+        resetPageAttributes();
+        store.infiniteScroll.addItemsToPage();
+
+        var searchQuery =  url.split("q=")[1];
+        $('#search').val(formatSearchQuery(decodeURIComponent(searchQuery)));
+    };
+
+    var resetPageAttributes = function(){
+        store.rows_added = 0;
+        store.last_to = 0;
+        store.items_per_row = 0;
+        store.doPagination = true;
+        store.firstRun = false;
+        store.infiniteScroll.recalculateRowsAdded();
+    };
+
+    $('#categorization :checkbox').click(function() {
+        var $this = $(this);
+        var data = {};
+        data.parent = $this.attr('name');
+        data.text = $this.attr('value');
+        // $this will contain a reference to the checkbox
+        if ($this.is(':checked')) {
+            // the checkbox was checked
+            triggerEvent(data, false);
+        } else {
+            // the checkbox was unchecked
+            triggerEvent(data, true);
+        }
+    });
+
+    $('div[data-toggle="collapse"]').on('click',function(){
+        var objectID=$(this).attr('href');
+        var icon = $(objectID).prev().find('.status').children();
+
+        if($(objectID).hasClass('in'))
+        {
+            $(objectID).collapse('hide');
+            icon.removeClass('fw-up');
+            icon.addClass('fw-down');
+        }
+
+        else{
+            $(objectID).collapse('show');
+            icon.removeClass('fw-down');
+            icon.addClass('fw-up');
+        }
+    });
+
+    var url = decodeURIComponent(window.location.href);
+    if(url.indexOf("=") > -1){
+        var query = url.split("=");
+        if(query[1] !== ""){
+            updateFilters(url);
+        }
+    }
+
+});
```

Comment:
Missing documentation. Better if you can explain the methods and important code lines.